SuperCalculator/src/supercalculator/Math.java

```diff
@@ -7,18 +7,111 @@
 
 /**
  *
- * @author za
+ * @author za, Haitham Abdel-Salam
  */
-public class Math {
-    
-    public static double factorial(int n){
-        
-        if(n == 0 || n == 1){
+public final class Math {
+
+    private Math() {
+
+    }
+    /**
+     *@PI Fixed PI value
+     */
+    public static final double PI = 3.141592653589793;
+
+    /**
+     *@PI_DEG Fixed PI value in degree
+     */
+    public static final double PI_DEG = 180;
+
+    /**
+     * Function calculates factorial of a givien integer
+     *@param n integer to calculate its factorial
+     */
+    public static double calculateFactorial(final int n) {
+        if (n == 0 || n == 1) {
             return 1.0d;
         }
-        
-        double result = n * factorial(n-1);
+
+        double result = n * calculateFactorial(n - 1);
         return result;
     }
-    
+
+    /**
+     * Converts Radian to Degree
+     *@param degreeValue value in degrees
+     */
+    public static double convertDegToRad(final double degreeValue) {
+        return degreeValue * (PI / PI_DEG);
+    }
+
+    /**
+     * Calculates power of a number
+     *@param value  number
+     * @param n power
+     */
+    public static double calculatePower(final double value, final double n) {
```

Comment:
This will only work for positive values of n (the exponent), so maybe change it to an integer to make sure it's safer. Would be ideal to also catch negative values of n and deal accordingly but that's lesser priority.